core/src/runtime/mod.rs

```diff
@@ -529,6 +536,29 @@ impl Runtime {
 
         let mut next_pc = self.state.pc.wrapping_add(4);
 
+        // Determine the category based on opcode for logging purposes
+        let category = match instruction.opcode {
```

Comment:
i don't really like this, feels not very extensible. instead, can you just keep track of Opcode => counts of how many time they appear in a HashMap?

all analysis like Arithmetic , Syscalls, etc can be done post hoc on the raw data in the execution report.